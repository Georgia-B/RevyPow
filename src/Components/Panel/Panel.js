import React from 'react';
import Temperature from '../TemperatureSection/Temperature';
import SnowSection from '../SnowSection/SnowSection';
import Loading from '../Loading/Loading';
import ContentSection from '../ContentSection/ContentSection';

const Panel = (props) => {
    const { data } = props;
    if (!data) {
        return <Loading />
    }
    return (
        <section className="panel">
            <div>
                <h1 className="panel__title">RevyPow</h1>
            </div>
            <ContentSection className="snow-container">
                <SnowSection data={data.newSnow} />
                <SnowSection data={data.base} />
            </ContentSection>
            <ContentSection>
                {data.temperatures.map(temp => {
                    return <Temperature
                        key={temp.title}
                        title={temp.title}
                        subtitle={temp.subtitle}
                        value={temp.value} />
                })}
            </ContentSection>
            <div>
                <div>Last updated</div><div>{data.dateUpdated.value}</div>
            </div>
        </section>
    );
}

export default Panel;
