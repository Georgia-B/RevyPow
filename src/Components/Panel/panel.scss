.panel {
    min-height: 60vh;
    width: 70%;
    margin: auto;
    padding: 20px;
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    border-radius: 5px;
    color: white;
    background-color: #4343435e;

    -webkit-animation: fadein 2s; /* Safari, Chrome and Opera > 12.1 */
    -moz-animation: fadein 2s; /* Firefox < 16 */
     -ms-animation: fadein 2s; /* Internet Explorer */
      -o-animation: fadein 2s; /* Opera < 12.1 */
         animation: fadein 2s;

    &__title {
        font-weight: lighter;
        font-size: 34px;
        margin-top: 0;
    }
}

@keyframes fadein {
 from { opacity: 0; }
 to   { opacity: 1; }
}
  
@media (max-width: $breakpoint-medium) {
    .panel {
        min-height: 75vh;
        width: 80%;
    }
}
  
@media (max-width: $breakpoint-small) {
    .panel {
      min-height: 80vh;
      width: 85%;
      padding: 0 10px;
    }
}