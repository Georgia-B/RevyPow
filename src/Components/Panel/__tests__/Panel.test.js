import React from 'react'
import { cleanup, render } from '@testing-library/react'
import Panel from '../Panel';


const testData = {
    temperatures: [
        {
            title: "Subpeak",
            subtitle: "2000m",
            value: "2"
        }
    ],
    wind: {
        title: "windspeed",
        value: "5"
    },
    base: {
        title: "Base snow",
        value: "40"
    },
    newSnow: {
        title: "Fresh snow",
        value: "2"
    },
    dateUpdated: {
        value: "yesterday"
    }
}

describe('Components | <Panel /> with no data', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<Panel />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})

describe('Components | <Panel /> with data', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<Panel data={testData} />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})