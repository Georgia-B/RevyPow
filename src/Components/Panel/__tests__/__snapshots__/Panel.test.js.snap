// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <Panel /> with data should render correctly 1`] = `
<DocumentFragment>
  <section
    class="panel"
  >
    <div>
      <h1
        class="panel__title"
      >
        RevyPow
      </h1>
    </div>
    <section
      class="content-section snow-container"
    >
      <div
        class="snowsection"
      >
        <h3
          class="snowsection__title"
        >
          Fresh snow
        </h3>
        <span
          class="snowsection__value"
        >
          2 cm
        </span>
      </div>
      <div
        class="snowsection"
      >
        <h3
          class="snowsection__title"
        >
          Base snow
        </h3>
        <span
          class="snowsection__value"
        >
          40 cm
        </span>
      </div>
    </section>
    <section
      class="content-section"
    >
      <div
        class="temperature"
      >
        <div>
          <h3
            class="temperature__title"
          >
            Subpeak
          </h3>
          <span>
            2000m
          </span>
        </div>
        <div
          class="temperature__value"
        >
          2 
          <span
            class="temperature__unit"
          >
            Â°C
          </span>
        </div>
      </div>
    </section>
    <div>
      <div>
        Last updated
      </div>
      <div>
        yesterday
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Components | <Panel /> with no data should render correctly 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <div
      class="loading__spinner"
    >
      <div
        class="loading__spinner__item"
      />
      <div
        class="loading__spinner__item"
      />
      <div
        class="loading__spinner__item"
      />
      <div
        class="loading__spinner__item"
      />
    </div>
  </div>
</DocumentFragment>
`;
